lib/jwt/ClientCapability.js

```diff
@@ -56,86 +53,84 @@ _.extend(IncomingClientScope.prototype, {
-    throw new Error('Required parameter "options" missing.');
-  }
-  if (_.isUndefined(options.applicationSid)) {
-    throw new Error('Required parameter "options.applicationSid" missing.');
-  }
-
-  options = options || {};
-  this.applicationSid = options.applicationSid;
-  this.clientName = options.clientName;
-  this.params = options.params;
-}
+class OutgoingClientScope {
+  constructor(options) {
+    if (!options) {
+      throw new Error('Required parameter "options" missing.');
+    }
+    if (!options.applicationSid) {
+      throw new Error('Required parameter "options.applicationSid" missing.');
+    }
 
-_.extend(OutgoingClientScope.prototype, {
-  scope: 'scope:client:outgoing',
+    options = options || {};
+    this.applicationSid = options.applicationSid;
+    this.clientName = options.clientName;
+    this.params = options.params;
+    this.scope = 'scope:client:outgoing';
+  }
 
-  payload: function() {
-    var queryArgs = [_.join(['appSid', qs.escape(this.applicationSid)], '=')];
+  payload() {
+    var queryArgs = [['appSid', qs.escape(this.applicationSid)].join('=')];
 
-    if (_.isString(this.clientName)) {
-      queryArgs.push(_.join(['clientName', qs.escape(this.clientName)], '='));
+    if (typeof this.clientName === 'string') {
+      queryArgs.push(['clientName', qs.escape(this.clientName)].join('='));
     }
 
-    if (_.isObject(this.params)) {
-      var queryParams = _.map(this.params, function(value, key) {
-        return _.join([qs.escape(key), qs.escape(value)], '=');
-      });
-      var filterParams = _.join(queryParams, '&');
+    if (typeof this.params === 'object') {
+      var queryParams = Object.entries(this.params).map(param => {
+        return [qs.escape(param[0]), qs.escape(param[1])].join('=');
+      })
+      var filterParams = queryParams.join('&');
 
-      queryArgs.push(_.join(['appParams', qs.escape(filterParams)], '='));
+      queryArgs.push(['appParams', qs.escape(filterParams)].join('='));
     }
 
-    var queryString = _.join(queryArgs, '&');
-    return _.join([this.scope, queryString], '?');
+    var queryString = queryArgs.join('&');
+    return [this.scope, queryString].join('?');
   }
-});
+}
 
 /**
  * @constructor
  * @param options
  */
-function ClientCapability(options) {
-  if (_.isUndefined(options)) {
-    throw new Error('Required parameter "options" missing.');
-  }
-  if (_.isUndefined(options.accountSid)) {
-    throw new Error('Required parameter "options.accountSid" missing.');
-  }
-  if (_.isUndefined(options.authToken)) {
-    throw new Error('Required parameter "options.authToken" missing.');
-  }
-
-  options = options || {};
-  this.accountSid = options.accountSid;
-  this.authToken = options.authToken;
-  this.ttl = options.ttl || 3600;
-  this.scopes = [];
-}
+class ClientCapability {
+  constructor(options) {
+    if (!options) {
+      throw new Error('Required parameter "options" missing.');
+    }
+    if (!options.accountSid) {
```

Comment:
Should these be checking against `undefined` instead of negating?

Reply:
`undefined` is falsy by default in JS, so negating it makes it truthy (https://stackoverflow.com/questions/39075142/javascript-undefined-gives-true)